packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -1,7 +1,96 @@
+import "@paprika/helpers/lib/polyfills/elementClosest";
 import useListBox from "../../../useListBox";
 import invokeOnChange from "../../../helpers/invokeOnChange";
 import { KEY_PRESS, CLICK } from "../../../types";
 
+function getOptions(event) {
+  let list = null;
+  if (event.target.dataset.pkaAnchor && event.target.dataset.pkaAnchor === "list-filter-input") {
+    // the event is coming from the filter
+    list = event.target.closest("[data-pka-anchor='list-box-box']").querySelector("[data-pka-anchor='styled-list']");
+  } else {
+    list = event.target.querySelector("[data-pka-anchor='styled-list']");
+  }
+
+  // filtering only results with role="option" attribute
+  return [...list.children].filter(li => li.hasAttribute("role"));
+}
+
+function getNextUp(event) {
+  function next(element) {
+    const list = event.target.closest("[data-pka-anchor='styled-list']");
+    const options = list.children;
+    if (list && [...options].indexOf(element) > 0) {
+      const sibling = element.previousElementSibling;
+      // if the next sibling has aria-hidden we skip it and call
+      // again previousSibling
+      if (sibling.hasAttribute("aria-hidden")) {
+        return next(sibling);
+      }
+
+      return sibling;
+    }
+
+    const box = document.activeElement.closest("[data-pka-anchor='list-box-box']");
+    const filter = box.querySelector("[data-pka-anchor='list-filter-input']");
+    if (filter) {
+      filter.focus();
+    }
+
+    return null;
+  }
+
+  // an activeElement is an option
+  if (document.activeElement.getAttribute("role") === "option") {
+    return next(document.activeElement);
+  }
+
+  // filter is already focussed just ignore the event
+  if (document.activeElement.dataset.pkaAnchor === "list-filter-input") {
```

Comment:
Do we need `.dataset?.` here?

Reply:
is not possible to reach this code if you haven't focus on the item first, therefore is certain that the object will be either a role="option" or a filter input both with a dataset property.